tests/rust/tcp-test/close/wait.rs

```diff
@@ -224,3 +227,119 @@ fn wait_after_async_close_connecting_socket(libos: &mut LibOS, remote: &SocketAd
 
     Ok(())
 }
+
+fn wait_on_invalid_queue_token_returns_einval(libos: &mut LibOS) -> Result<()> {
+    println!("{}", stringify!(wait_on_invalid_queue_token_returns_einval));
+
+    // Wait on an invalid queue token.
+    match libos.wait(QToken::from(u64::MAX), Some(Duration::from_micros(0))) {
+        Ok(_) => anyhow::bail!("wait() should not succeed on invalid token"),
+        Err(e) if e.errno == libc::EINVAL => {},
+        Err(_) => anyhow::bail!("wait() should not fail with any other reason than invalid token"),
+    }
+
+    Ok(())
+}
+
+fn wait_for_accept_after_issuing_async_close(libos: &mut LibOS, local: &SocketAddrV4) -> Result<()> {
+    println!("{}", stringify!(wait_for_accept_after_issuing_async_close));
+
+    // Create an accepting socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+    libos.bind(sockqd, *local)?;
+    libos.listen(sockqd, 16)?;
+    let qt: QToken = libos.accept(sockqd)?;
+
+    // Poll once to ensure that the accept() co-routine runs.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Err(e) if e.errno == libc::ETIMEDOUT => {},
+        // If we found a connection to accept, something has gone wrong.
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_ACCEPT && qr.qr_ret == 0 => {
+            anyhow::bail!("accept() should not succeed because remote should not be connecting")
+        },
+        Ok(_) => anyhow::bail!("wait() should not succeed with accept()"),
+        Err(_) => anyhow::bail!("wait() should timeout with accept()"),
+    }
+
+    let qt_close: QToken = libos.async_close(sockqd)?;
+
+    // Wait again on accept and ensure that ETIMEDOUT is returned.
```

Comment:
Should you reword `accept` to `accept()`?

Reply:
Done.